src/main/kotlin/dqualizer/dqexec/adapter/StimulusAdapter.kt

```diff
@@ -148,35 +154,44 @@ class StimulusAdapter(private val loadTestConstants: LoadTestConstants) {
     /**
      * Create an ordered set of stages for the 'INCREASE_LOAD' load_profile
      *
-     * @param stageDuration The duration of one single stage
-     * @param startTarget   The amount of users at the first stage
-     * @param endTarget     The amount of users at the last stage
-     * @param exponent      How fast should the amount of users increase
+     * @param startTarget The amount of users at the first stage
+     * @param endTarget   The amount of users at the last stage
+     * @param exponent    How fast should the amount of users increase
      * @return An ordered set of stages
      */
     private fun getIncreasingStages(
-        stageDuration: String,
-        startTarget: Int,
-        endTarget: Int,
-        exponent: Int
+        startTarget: Int, endTarget: Int, exponent: Int,
+        testDurationString: String, numberOfStages: Int
     ): LinkedHashSet<Stage> {
         val stages = LinkedHashSet<Stage>()
-        var stageCounter = 1
-        var currentTarget = startTarget
-
-        while (currentTarget < endTarget) {
-            currentTarget = startTarget * Math.pow(stageCounter.toDouble(), exponent.toDouble()).toInt()
-            val currentStage = Stage(stageDuration, currentTarget)
+        val stageDurationString: String = computeStageDurationString(testDurationString, numberOfStages)
+        val loadCurve = LoadCurveHelper(exponent.toDouble(), startTarget.toDouble(), endTarget.toDouble(), numberOfStages.toDouble())
+        val firstStage = Stage("0s", startTarget)
+        stages.add(firstStage)
+        for (i in 0 until numberOfStages) {
+            val endTimeOfStage = i + 1
+            val currentStage = Stage(
+                stageDurationString,
+                loadCurve.evaluate(endTimeOfStage.toDouble()).roundToInt()
+            )
             stages.add(currentStage)
-
-            stageCounter++
         }
-        val lastStage = Stage(stageDuration, 0)
+        val lastStage = Stage(stageDurationString, 0)
         stages.add(lastStage)
-
         return stages
     }
 
+
+    private fun computeStageDurationString(testDurationString: String, numberOfStages: Int): String {
+        val testDuration = convertToTimeString(testDurationString)
+        val testDurationSeconds = testDuration!!.seconds
```

Comment:
This `!!` is redundant if the suggestion for `convertToTimeString` below is applied.

```suggestion
        val testDurationSeconds = testDuration.seconds
```